dtfs-central-api/src/v1/controllers/portal/gef-deal/add-min-activities.controller.js

```diff
@@ -0,0 +1,192 @@
+    facilityType: facility ? `${facility.type} facility` : '',
+    facilityID: facility ? facility.ukefFacilityId : '',
+    maker,
+    checker,
+  };
+
+  return portalActivityObj;
+};
+
+/**
+ * For facilities changed to issued
+ * adds to front of portalActivity array in correct format
+ * @param {Object} application
+ * @param {Array} facilities
+ * @returns {Array} portalActivities
+ */
+const facilityChangePortalActivity = async (application, facilities) => {
+  const { checkerId, portalActivities } = application;
+  const checker = await getUserInfo(checkerId);
+
+  facilities.forEach(async (facility) => {
+    if (facility.canResubmitIssuedFacilities) {
+      // creates user object to add to array
+      const maker = facility.unissuedToIssuedByMaker;
+      const activityParams = {
+        type: PORTAL_ACTIVITY_LABEL.FACILITY_CHANGED_ISSUED,
+        user: '',
+        activityType: PORTAL_ACTIVITY_TYPE.FACILITY_STAGE,
+        activityText: '',
+        activityHTML: 'facility',
+        facility,
+        maker,
+        checker,
+      };
+      // generates an activities object
+      const activityObj = portalActivityGenerator(activityParams);
+      // adds to beginning of portalActivities array so most recent displayed first
+      portalActivities.unshift(activityObj);
+    }
+  });
+
+  return portalActivities;
+};
+
+/**
+ * Generates activity for MIN submission to UKEF
+ * Adds to front of portalActivities array in correct format
+ * @param {Object} application
+ * @returns {Array} portalActivities
+ */
+const ukefSubmissionPortalActivity = async (application) => {
+  const { portalActivities, checkerId } = application;
+
+  // generates the label for activity array
+  const applicationType = PORTAL_ACTIVITY_LABEL.MIN_SUBMISSION;
+  // creates user object to add to array
+  const user = await getUserInfo(checkerId);
+  const activityParams = {
+    type: applicationType,
+    user,
+    activityType: PORTAL_ACTIVITY_TYPE.NOTICE,
+    activityText: '',
+    activityHTML: '',
+    facility: '',
+    maker: '',
+    checker: '',
+  };
+  // generates an activities object
+  const activityObj = portalActivityGenerator(activityParams);
+  // adds to beginning of portalActivities array so most recent displayed first
+  portalActivities.unshift(activityObj);
+
+  return portalActivities;
+};
+
+/**
+ * Generates MIN Activity objects for submission and changed facilities
+ * Removes flag for changedToIssued facility
+ * Returns updated deal
+ * @param {*} req
+ * @param {*} res
+ */
+const generateMINActivities = async (req, res) => {
+  const dealId = req.params.id;
+
+  const application = await findOneDeal(dealId);
+
+  if (application) {
+    const facilities = await findAllGefFacilitiesByDealId(dealId);
+    let { portalActivities } = application;
+
+    portalActivities = await ukefSubmissionPortalActivity(application);
+    portalActivities = await facilityChangePortalActivity(application, facilities);
```

Comment:
I think we need some renaming here, or we need to declare a different variable. I'm not sure I understand why `portalActivities` gets overwritten twice

Reply:
because it's adding 2 different items to the same array from applications and keeping what is already in there